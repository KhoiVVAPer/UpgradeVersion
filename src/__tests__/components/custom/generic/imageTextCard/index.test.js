import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ImageTextCard from '../../../../../components/custom/generic/imageTextCard';

const setUp = (props) => {
  const component = shallow(<ImageTextCard {...props} />);
  return component;
};

describe('components > custom > elements > ImageTextCard', () => {
  let component;
  beforeEach(() => {
    const props = {
      headline: 'heading',
      text: '<p>description</p>',
      url: 'https://s1.kaercher-media.com/products/13171300/main/1/d3.jpg'
    };
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders image component', () => {
    const input = component.find({ dcaTest: 'image' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders heading', () => {
    const input = component.find({ dcaTest: 'heading' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders description', () => {
    const input = component.find({ dcaTest: 'description' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });
});
