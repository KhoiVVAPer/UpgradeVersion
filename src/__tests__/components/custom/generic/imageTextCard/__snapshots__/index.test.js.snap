// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > ImageTextCard Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock />
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            heading
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            <react-native-mock>
              description
            </react-native-mock>
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
